### PR TITLE
fix(core): overflow controller scroll on small screens

### PR DESCRIPTION
**Pull request by zeroedin**
## What I did

1. Refactored scroll calculation for scrollLeft and scrollRight in overflow controller
2. Added smooth scroll behavior
3. Refactored showScrollButtons calculation to include scroll button width in hiding and showing the buttons.

## Testing Instructions

1. View [Deploy preview ](https://deploy-preview-2486--patternfly-elements.netlify.app/components/tabs/demo/)

## Notes to Reviewers

1. Ensure that this works across all browsers and screen sizes.  Please use Browserstack and try a variety of devices

### CODE REVIEW

**Review by nikkimk — COMMENTED** [review]
Looks good, except for that tabs breaks in Safari versions older than 14.

**Review by nikkimk — APPROVED** [review]
TL;DR: LGTM

So the issue I found on Safari is also an issue on https://patternflyelements.org/components/tabs/demo but not on the [DP for #2845 ](https://deploy-preview-2485--patternfly-elements.netlify.app/components/tabs/demo/).

This tells me that this PR doesn't break Safari and that the other PR should fix it.

**Review by heyMP — APPROVED** [review]
Smooth! 👍
